### PR TITLE
Add a script to detect whether go files contain copyright headers

### LINKED ISSUE #33: Check copyright header in Travis
We should check the copyright header in Travis CI to avoid accidentally committing new files without the headers.

### PR DESCRIPTION
**Pull request by lookuptable**
This closes #33 .

### CODE REVIEW

File: bin/headers.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+failure=0
```
Comment (wlu2016) [inline bin/headers.sh]: failure -> failures

File: bin/headers.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+for file in $(find . -name "*.go"); do
+  head -n 1 $file | grep '^// Copyright 2017 Istio Authors' > /dev/null
```
Comment (wlu2016) [inline bin/headers.sh]: remove "Copyright 2017"? or use a regex to match "Copyright any_year Istio Authors"

File: bin/headers.sh (hunk 1 above)
Comment (lookuptable) [inline bin/headers.sh]: This is a flag indicating whether there's any error. I'll change it to something like `has_failure` to make it more meaningful.

File: bin/headers.sh (hunk 2 above)
Comment (lookuptable) [inline bin/headers.sh]: Done

**Pull request merged**